### PR TITLE
Add some basic hooks to the new hook library

### PR DESCRIPTION
**Pull request by whisperity**

### CODE REVIEW

File: www/scripts/codecheckerviewer/ListOfBugs.js [hunk 1]
```diff
@@ -117,8 +131,21 @@ function (declare, Deferred, ObjectStore, Store, QueryResults, topic,
               if (reportDataList instanceof RequestFailed)
                 deferred.reject(
                   'Failed to get reports: ' + reportDataList.message);
-              else
+              else {
                 deferred.resolve(that._formatItems(reportDataList));
+
+                if (query.reportFilters.length > 1)
```
Comment (csordasmarton) [inline www/scripts/codecheckerviewer/ListOfBugs.js]: You use this code part in three different places:
- https://github.com/whisperity/CodeChecker/blob/23639a20fdc5eef364a83dee1a162ae0ec8c841b/www/scripts/codecheckerviewer/ListOfBugs.js#L137-L147
- https://github.com/whisperity/CodeChecker/blob/23639a20fdc5eef364a83dee1a162ae0ec8c841b/www/scripts/codecheckerviewer/ListOfBugs.js#L167-L177
- https://github.com/whisperity/CodeChecker/blob/23639a20fdc5eef364a83dee1a162ae0ec8c841b/www/scripts/codecheckerviewer/ListOfBugs.js#L197-L207

I think you should create a function for this.

Comment (whisperity) [inline www/scripts/codecheckerviewer/ListOfBugs.js]: Good catch. Refactored.

File: www/scripts/codecheckerviewer/ListOfBugs.js [hunk 2]
```diff
@@ -57,8 +58,21 @@ function (declare, Deferred, ObjectStore, Store, QueryResults, topic,
...
             deferred.reject('Failed to get reports: ' + reportDataList.message);
-          else
+          else {
             deferred.resolve(that._formatItems(reportDataList));
+
+            if (query.reportFilters.length > 1)
+              hooks.bugFilteringChanged(query.reportFilters.length);
+            else {
+              var onlyFilter = query.reportFilters[0];
+              if (onlyFilter.filepath != "**" || onlyFilter.checkerId ||
```
Comment (csordasmarton) [inline www/scripts/codecheckerviewer/ListOfBugs.js]: Use `!==` operator for equality check.
Moreover you use these condition in multiple places:
- https://github.com/whisperity/CodeChecker/blob/d403e6da6ed3cf7a089e0892af86b98c05ee823d/www/scripts/codecheckerviewer/ListOfBugs.js#L68-L70
- https://github.com/whisperity/CodeChecker/blob/d403e6da6ed3cf7a089e0892af86b98c05ee823d/www/scripts/codecheckerviewer/ListOfBugs.js#L125-L127

File: www/scripts/codecheckerviewer/hooks.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    bugFilteringChangedDiff : function (numFilters) {
+    },
+
+    bugReportOpened : function (report) {
+    },
+
+    bugPathEventClicked : function (index, bugPathEvent) {
+    }
+  };
+});
```
Comment (bruntib) [inline www/scripts/codecheckerviewer/hooks.js]: This should be implemented by `dojo/topic` which provides this message sending functionality. If you browse the source code, you can find `codechecker/...` topics already registered. And personally I think that this interface shouldn't be added to the code base without implementation just for future use.

File: www/scripts/codecheckerviewer/ListOfBugs.js [hunk 4]
```diff
@@ -57,8 +58,21 @@ function (declare, Deferred, ObjectStore, Store, QueryResults, topic,
         function (reportDataList) {
           if (reportDataList instanceof RequestFailed)
             deferred.reject('Failed to get reports: ' + reportDataList.message);
-          else
+          else {
             deferred.resolve(that._formatItems(reportDataList));
+
+            if (query.reportFilters.length > 1)
```
Comment (gyorb) [inline www/scripts/codecheckerviewer/ListOfBugs.js]: Could we create one common filter hook?
This part is almost the same as the `filterHook` function later.

Comment (whisperity) [inline www/scripts/codecheckerviewer/ListOfBugs.js]: Of course. I have refactored the method out.

**Pull request merged**